orient_express/sklearn_pipeline.py

```diff
@@ -0,0 +1,80 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class LabelEncoderTransformer(BaseEstimator, TransformerMixin):
+    """
+    A wrapper class that integrates a model and a label encoder to handle
+    encoded predictions and their corresponding probabilities.
+
+    Attributes:
+        model (object): A machine learning model with `fit`, `predict`, and `predict_proba` methods.
+        label_encoder (object): A label encoder that maps encoded class labels to their original string labels.
+
+    Methods:
+        fit(X, y):
+            Trains the model on the provided features and labels.
+        predict(X):
+            Predicts the classes for the given features and returns the original string labels.
+        predict_proba(X):
+            Returns the probabilities for each class for the given features, along with their original class labels.
+    """
+
+    def __init__(self, model, label_encoder):
+        """
+        Initializes the LabelEncoderTransformer.
+
+        Args:
+            model (object): A machine learning model.
+            label_encoder (object): A label encoder with `fit` and `inverse_transform` methods.
+        """
+        self.model = model
+        self.label_encoder = label_encoder
+
+    def fit(self, X, y):
+        """
+        Fits the model to the training data.
+
+        Args:
+            X (array-like): Feature matrix.
+            y (array-like): Target labels.
+
+        Returns:
+            self: Fitted LabelEncoderTransformer instance.
+        """
+        self.model.fit(X, y)
+        return self
+
+    def predict(self, X):
+        """
+        Predicts the target labels and returns the original string labels.
+
+        Args:
+            X (array-like): Feature matrix for predictions.
+
+        Returns:
+            array-like: Predicted class labels in their original form.
+        """
+        encoded_predictions = self.model.predict(X)
+        return self.label_encoder.inverse_transform(encoded_predictions)
```

Comment:
Why do we need an inverse_transform here?

Reply:
label encoder can encode labels into ids and back. That's necessary, since the model can't usually operate with labels directly, and require a preprocessing step.
That adds inconvenience for model use, since the models spit out the ids, and we need to make a backward conversion in the end. This PR is all about making this inverse transform after the prediction is done, and allow the pipeline to spit out string labels.